modules/swagger-codegen/src/main/resources/php/api.mustache

```diff
@@ -181,8 +181,12 @@ use {{invokerPackage}}\ObjectSerializer;
         {{#responses}}
             {{#dataType}}
                 {{^isWildcard}}case {{code}}:{{/isWildcard}}{{#isWildcard}}default:{{/isWildcard}}
+                    $content = $e->getResponseBody();
+                    if ('{{dataType}}' !== 'string') {
```

Comment:
@ymilin what about other primitive types such as integer, boolean, etc? Do we need to check those as well?

Reply:
@wing328 Thanks for the review 😄

I went ahead and took a better look at the [`ObjectSerializer.mustache::deserialize`](https://github.com/swagger-api/swagger-codegen/blob/432b3589861d4a35e2d4a7a1f6a55b889645886f/modules/swagger-codegen/src/main/resources/php/ObjectSerializer.mustache#L218
) function. This PR seems to effectively break other primitive type support (especially with booleans as they get decoded as (int) 1 and null for true and false respectively).

Now it feels to me that the json_decode part should be handled in deserialize as the function has knowledge of both the data and its type.

The same approach is already used with "success" responses and is quite concerning for the same reasons, here 

https://github.com/swagger-api/swagger-codegen/blob/432b3589861d4a35e2d4a7a1f6a55b889645886f/modules/swagger-codegen/src/main/resources/php/api.mustache#L159-167

I'll spend some time and test with all the types from the OpenAPI spec and validate all this.

Reply:
@wing328 I was trying to resolve the same problem and found out that the approach that was used here is correct, because php `json_encode` function properly handle raw types, e.g.:
```php
echo "json_decode('123')\n";
var_dump(json_decode('123'));

echo "json_decode('123.45')\n";
var_dump(json_decode('123.45'));

echo "json_decode('true')\n";
var_dump(json_decode('true'));

echo "json_decode('false')\n";
var_dump(json_decode('false'));
```

will return:
```
json_decode('123')
int(123)
json_decode('123.45')
double(123.45)
json_decode('true')
bool(true)
json_decode('false')
bool(false)
```
So each primitive type handled by `json_decode` properly (I think the same idea was behind handling success response with the same check only for string primitive type). I also generate client and test it with real API and got correct answers.

I believe it's safe to apply theses changes:
```php
$content = $e->getResponseBody();
if ('{{dataType}}' !== 'string') {
	$content = json_decode($content);
}
$data = ObjectSerializer::deserialize(
	$content,
	'{{dataType}}',
	$e->getResponseHeaders()
);
```

Is it possible to accept this PR or it is necessary  to do some extra actions? E.g. update code or create a new PR?

Reply:
Hi, this PR was ported and merged to the [OpenAPITools/openapi-generator](https://github.com/OpenAPITools/openapi-generator) fork of this project.

Link to the PR over there : [https://github.com/OpenAPITools/openapi-generator/pull/757](https://github.com/OpenAPITools/openapi-generator/pull/757)